src/compass/utils/h5_helpers.py

```diff
@@ -79,6 +79,33 @@ def add_dataset_and_attrs(group, meta_item):
         val_ds.attrs[key] = _as_np_string_if_needed(val)
 
 
+def _determine_fill_value(dtype):
+    '''
+    Helper function to determine COMPASS specific fill values based on h5py
+    Dataset type (dtype)
+    '''
+    # Possible float types and float fill
+    float_types = [np.double, np.single, np.float32, np.float64, 'float32',
```

Comment:
Why do we differentiate between `np.float32` and `float32`?

Reply:
For its `dtype` parameter, `require_dataset` can take  the string 'float32' and class/type `np.float32`. We use both in our code.